docs(cli): add Local Proxy page (run multiple shops in parallel) (#2444)
shopware/docs: products/tools/cli/project-commands/local-proxy.md

    B->>R: where is my-shop.shopware.local?
    R->>D: routed locally
    D-->>B: 127.0.0.1
    B->>T: https://my-shop.shopware.local
    T->>S: routed by hostname
    S-->>B: response
```

Nothing leaves your machine: the DNS resolver only ever answers `127.0.0.1`, and shops reach each other by the same trusted hostnames.

## One-time setup

Run once per machine. It points the operating system's resolver at the local DNS and installs the certificate authority into your trust stores — the single `sudo` moment:

```bash
shopware-cli project proxy setup
```

To use a different base domain (persisted machine-wide), pass `--domain`:

```bash
shopware-cli project proxy setup --domain shopware.test
```

## Run a shop through the proxy

### New project

The simplest path is interactive — create the project and answer the prompts:

```bash
shopware-cli project create my-shop
```

`project create` asks whether to use **local domains** and offers to run the one-time machine setup for you (the `sudo` step). Answer yes, then start the shop — it comes up at `https://my-shop.shopware.local` with no further steps:

```bash
cd my-shop
shopware-cli project dev
```

> For scripting or CI, skip the prompts with flags (the machine setup must have been run once beforehand):

Reviewer: ```suggestion
For scripting or CI, skip the prompts with flags (the one-time machine setup must already have been completed):
```

```bash

> shopware-cli project create --local-domain my-shop

Reviewer: Don't you have to use --no-interaction flag as well?

Author: ```bash
shopware-cli project create --local-domain my-shop
```

That's interactive way of creating new shop with explicit argument for local domain (read: proxy).

Author: `--no-interaction` would mean non interactive shop creation

Reviewer: Thought for CI etc. the --no-interaction would still be needed, because otherwise it would try to start the interactive shop creation?

Reviewer: (In other words: example shows how to create the local domain without interactions, but it does not fit the description above with CI use)

Author: Now I see it, good catch!

```

### Existing project

> Opt an existing port-based project in, and revert it when you are done:

Reviewer: ```suggestion
Add an existing port-based project to the proxy, and revert it when you are done:
```
